recipes/bimg/all/conanfile.py

```diff
@@ -0,0 +1,251 @@
+            genie_extra += " --with-dynamic-runtime"
+        if self.options.tools:
+            genie_extra += " --with-tools"
+        return genie_extra
+
+    @property
+    def _lib_target_prefix(self):
+        if self.settings.os == "Windows":
+            return "libs\\"
+        else:
+            return ""
+
+    @property
+    def _tool_target_prefix(self):
+        if self.settings.os == "Windows":
+            return "tools\\"
+        else:
+            return ""
+
+    @property
+    def _projs(self):
+        projs = [f"{self._lib_target_prefix}bimg", f"{self._lib_target_prefix}bimg_decode", f"{self._lib_target_prefix}bimg_encode"]
+        if self.options.tools:
+            projs.extend([f"{self._lib_target_prefix}texturec"])
+        return projs
+
+    @property
+    def _compiler_required(self):
+        return {
+            "gcc": "8",
+            "clang": "3.3",
+            "apple-clang": "5",
+        }
+
+    @property
+    def _bx_version(self): #mapping of bimg version to required/used bx version
+        return {"cci.20230114": "cci.20221116"}
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires(f"bx/{self._bx_version[self.version]}@")
+
+    def validate(self):
+        if not self.options.get_safe("fPIC", True):
+            raise ConanInvalidConfiguration("This package does not support builds without fPIC.")
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 14)
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            try:
+                minimum_required_compiler_version = self._compiler_required[str(self.settings.compiler)]
+                if Version(self.settings.compiler.version) < minimum_required_compiler_version:
+                    raise ConanInvalidConfiguration("This package requires C++14 support. The current compiler does not support it.")
+            except KeyError:
+                self.output.warn("This recipe has no checking for the current compiler. Please consider adding it.")
+
+    def build_requirements(self):
+        self.tool_requires("genie/1170")
+        if not is_msvc(self) and self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True,
+                    destination=os.path.join(self.source_folder, self._bimg_folder))
+        #We need bx source too
+        get(self, **self.dependencies["bx"].conan_data["sources"][self._bx_version[self.version]], strip_root=True,
```

Comment:
Is it possible to patch instead and consume bx as package instead?

Reply:
It would be very difficult as the entire build system (genie project) relies on it being that way, and I'm not sure if I'm good enough with genie to modify it to do it that way instead. I could try to play around with doing that but I think it might be too much work (with even more in the future to keep it working) than it would be worth, considering it wouldn't actually affect projects that just use bgfx/bimg/bx and that grabbing bx source from the dependency package when building is easy and straightforward.

Reply:
I remember about the discussion using genie in another PR. If say it's very difficult, please, keep doing dual download, but add a comment in the recipe explaining it. So far, this build system is one of most exotics that we have in CCI.

Reply:
The fact that bimg needs bx source (and bgfx needs bimg and bx source) isn't really genie's fault, but it is how the genie projects for them are set up. 
The initial reason was that bx and bimg were basically part of bgfx and they were only split out as separate projects when they started getting bigger, and it was decided to just need their source to make it easy for people to build bgfx (and bimg and bx) without dealing with managing dependencies, just by also cloning bimg and bx in the same place. This was, of course, back when dependency management didn't have many or any good options like conan.
Perhaps in the future this can be changed, maybe even by using conan, but at the moment there aren't really any plans for it upstream. I will add a comment explaining this for now, thanks for the feedback!